### PR TITLE
SITL: break up GPS backends into separate files

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/SITL/SIM_GPS_GSOF.h
[the remark this answers is not in the document]
Comment (peterbarker) [reply libraries/SITL/SIM_GPS_GSOF.h:1]: Done

File: libraries/SITL/SIM_GPS.h [hunk 1]
```diff
@@ -327,16 +96,30 @@ class GPS : public SerialDevice {
...
 #endif
+#if AP_SIM_GPS_NOVA_ENABLED
         NOVA  =  8,
+#endif
+#if AP_SIM_GPS_SBP2_ENABLED
         SBP2  =  9,
-        GSOF  = 11, // matches GPS_TYPE
+#endif
+#if AP_SIM_GPS_TRIMBLE_ENABLED
+        TRIMBLE  = 11, // matches GPS_TYPE
```
Comment (Ryanf55) [inline libraries/SITL/SIM_GPS.h:118]: Trimble sells multiple types of GPS; one of them communicates GSOF protocol. 

I'd propose calling it `TRIMBLE_GSOF` that way it doesn't conflict with others like `TRIMBLE_TSIP` which is another protocol used in the larger GNSS systems they sell.

File: libraries/SITL/SIM_GPS_Trimble.h [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    // The application file record transmission number
+    uint8_t appfile_trans_num;
+
+    uint8_t payload[MAX_PAYLOAD_SIZE];
+
+    // Clear all parser state/flags for handling a fresh packet.
+    void reset();
+};
+
+class GPS_Trimble : public GPS_Backend, public DCOL_Parser {
```
Comment (Ryanf55) [inline libraries/SITL/SIM_GPS_Trimble.h:127]:
```suggestion
class GPS_Trimble_GSOF : public GPS_Backend, public DCOL_Parser {
```
What do you think about that name, so it doesn't conflict with other [Trimble GNSS systems like the NAV900 which uses TSIP protocol](https://xdevs.com/doc/Trimble/sveight/sv8_app_a.pdf)?

**Review by Ryanf55 — COMMENTED** [review]
Perhaps the rename of the GSOF parser class to `Trimble` was a bit much; I'm worried it will conflict with other products they sell. I've proposed some alternative names that will be conflict-free. 

Otherwise, I love the PR. Great to see!

**Review by tridge — APPROVED** [review]

**Pull request merged**